cfme/v2v/migration_plans.py

```diff
@@ -0,0 +1,635 @@
+
+    name = attr.ib()
+
+    def is_plan_in_progress(self):
+        """MIQ V2V UI is going through redesign as OSP will be integrated."""
+        view = navigate_to(self, "InProgress")
+
+        def _in_progress():
+            try:
+                is_plan_visible = view.progress_card.is_plan_visible(self.name)
+                if is_plan_visible:
+                    try:
+                        plan_time_elapsed = view.progress_card.get_clock(self.name)
+                        new_msg = "time elapsed for migration: {time}".format(
+                            time=plan_time_elapsed
+                        )
+                    except NoSuchElementException:
+                        new_msg = "playbook is executing.."
+                        pass
+                    logger.info(
+                        "For plan {plan_name}, is plan in progress: {visibility}, {message}".format(
+                            plan_name=self.name, visibility=is_plan_visible, message=new_msg
+                        )
+                    )
+                return not is_plan_visible
+            except ItemNotFound:
+                return True
+
+        return wait_for(
+            func=_in_progress,
+            message="migration plan is in progress, be patient please",
+            delay=5,
+            num_sec=1800,
+        )
+
+    def is_migration_complete(self):
+        """Uses search box to find migration plan, return True if found.
+        Args:
+            migration_plan name: (object) Migration Plan name
+        """
+        view = navigate_to(self, "Complete")
+        view.wait_displayed()
+        if self.appliance.version >= "5.10":
+            view.items_on_page.item_select("15")
+            view.search_box.fill("{}\n\n".format(self.name))
+        return (self.name in view.plans_completed_list.read() and
+                view.plans_completed_list.is_plan_succeeded(self.name))
+
+    def migration_plan_request(self):
+        view = navigate_to(self, "InProgress")
+        view.progress_card.select_plan(self.name)
+        request_view = self.create_view(MigrationPlanRequestDetailsView, wait="10s")
+        request_details_list = request_view.migration_request_details_list
+        view.items_on_page.item_select("15")
+        self.is_plan_in_progress()
+        return request_details_list
+
+
+@attr.s
+class MigrationPlanCollection(BaseCollection):
+    """Collection object for migration plan object"""
+
+    ENTITY = MigrationPlan
+
+    def create(
+        self,
+        name,
+        infra_map,
+        vm_list,
+        description=None,
+        csv_import=False,
+        target_provider=None,
+        osp_security_group=None,
+        osp_flavor=None,
+        pre_playbook=None,
+        post_playbook=None,
+        start_migration=False,
+    ):
+        """Create new migration plan in UI
+        Args:
+            name: (string) plan name
+            description: (string) plan description
```

Comment:
These look like they should be attributes of `MigrationPlan`, not just passed to the create method of the collection.